test/chart/AccessibilityLayer.spec.tsx

```diff
@@ -1,546 +1,688 @@
+            <YAxis />
+          </AreaChart>
+        </div>
+      );
+    };
+
+    test('When chart updates, arrow keys still work', () => {
+      const { container } = render(<Expand />);
 
-    fireEvent.keyDown(svg, {
-      key: 'ArrowRight',
-    });
-    expect(tooltip).toHaveTextContent('Page E');
+      const pre = container.querySelector('pre');
+      assertNotNull(pre);
+      const svg = container.querySelector('svg');
+      assertNotNull(svg);
+      const tooltip = getTooltip(container);
+
+      expect(tooltip.textContent).toBe('');
+      expect(pre.textContent).toBe('6');
 
-    // The chart only goes from A - E, so we shouldn't be able to go right any further.
-    fireEvent.keyDown(svg, {
-      key: 'ArrowRight',
+      // Once the chart receives focus, the tooltip should display
+      act(() => svg.focus());
+      expect(tooltip).toHaveTextContent('Page A');
+
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page B');
+
+      // Page C
+      arrowRight(svg);
+
+      // Page D
+      arrowRight(svg);
+
+      // Page E
+      arrowRight(svg);
+
+      // Page F
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page F');
+
+      fireEvent.click(container.querySelector('#my3') as HTMLButtonElement);
+      expect(pre.textContent).toBe('3');
+
+      // The chart only goes from A - C now, so the AccessibilityManager should think "C" is active.
+
+      arrowLeft(svg);
+      expect(tooltip).toHaveTextContent('Page B');
+
+      fireEvent.click(container.querySelector('#my5') as HTMLButtonElement);
+      expect(pre.textContent).toBe('5');
+
+      // The chart now goes from A - E. Since the focus was already on B, B can remain active.
+      expect(tooltip).toHaveTextContent('Page B');
+
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page C');
+
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page D');
+
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page E');
+
+      // The chart only goes from A - E, so we shouldn't be able to go right any further.
+      arrowRight(svg);
+      expect(tooltip).toHaveTextContent('Page E');
+    });
+
+    const Counter = () => {
+      const [count, setCount] = useState(0);
+
+      return (
+        <div>
+          <h1>
+            This is:
+            {count}
+          </h1>
+
+          <button type="button" onClick={() => setCount(count + 1)}>
+            Bump counter
+          </button>
+
+          <AreaChart width={100} height={50} data={PageData} accessibilityLayer={accessibilityLayer}>
+            <Area type="monotone" dataKey="uv" stroke="#ff7300" fill="#ff7300" />
+            <Tooltip />
+            <Legend />
+            <XAxis dataKey="name" />
+            <YAxis />
+          </AreaChart>
+        </div>
+      );
+    };
+
+    // Q: What is the point of this test?
+    // A: Well, I was thinking, what was a way this could break and would be hard to figure out?
```

Comment:
Cool! Future us thanks you

Reply:
(Wasn't my code, I only added indentation. I think Julianna added this one.)

Reply:
Oh hah, either way